core/src/main/java/hudson/util/io/ZipArchiver.java

```diff
@@ -81,19 +83,20 @@ public void visit(final File f, final String _relativePath) throws IOException {
         // by forward-slashes (/)
         String relativePath = _relativePath.replace('\\', '/');
 
-        if (f.isDirectory()) {
+        BasicFileAttributes basicFileAttributes = Files.readAttributes(f.toPath(), BasicFileAttributes.class);
```

Comment:
Careful! `File#toPath` can throw `InvalidPathException`. If callers aren't already handling `InvalidPathException` (which is often the case for methods that rethrow `IOException` rather than catching and handling `IOException | InvalidPathException `), better to use `Util.fileToPath`.

Reply:
Thanks for the tip, adjusted as suggested.